_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -1465,13 +1459,17 @@ mfxStatus VAAPIEncoder::CreateAuxilliaryDevice(
     };
 
     std::vector<VAConfigAttrib> attrs;
-    attrs.reserve(sizeof(attr_types) / sizeof(attr_types[0]));
+    auto AV = [&attrs, &idx_map](VAConfigAttribType t) -> uint32_t& { return attrs[idx_map[t]].value; };
 
-    for (size_t i=0; i < sizeof(attr_types)/sizeof(attr_types[0]); ++i)
+    mfxI32 i = 0;
```

Comment:
It's still better to "reserve" enough capacity as we know attrs final size beforehand.